File: binding/src/opengl/libqopengltimerquery.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qopengltimerquery.html
pub const qopengltimerquery = struct {

    /// New constructs a new QOpenGLTimerQuery object.
    pub fn New() QtC.QOpenGLTimerQuery {
        return qtc.QOpenGLTimerQuery_new();
    }


    /// New2 constructs a new QOpenGLTimerQuery object.
    pub fn New2(parent: ?*anyopaque) QtC.QOpenGLTimerQuery {
        return qtc.QOpenGLTimerQuery_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimerQuery_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimerquery.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Create(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimerQuery_Create(@ptrCast(self));
    }

    pub fn Destroy(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimerQuery_Destroy(@ptrCast(self));
    }

    pub fn IsCreated(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimerQuery_IsCreated(@ptrCast(self));
    }

    pub fn ObjectId(self: ?*anyopaque, ) u32 {
        return qtc.QOpenGLTimerQuery_ObjectId(@ptrCast(self));
    }

    pub fn Begin(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimerQuery_Begin(@ptrCast(self));
    }

    pub fn End(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimerQuery_End(@ptrCast(self));
    }

    pub fn WaitForTimestamp(self: ?*anyopaque, ) u64 {
        return qtc.QOpenGLTimerQuery_WaitForTimestamp(@ptrCast(self));
    }

    pub fn RecordTimestamp(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimerQuery_RecordTimestamp(@ptrCast(self));
    }

    pub fn IsResultAvailable(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimerQuery_IsResultAvailable(@ptrCast(self));
    }

    pub fn WaitForResult(self: ?*anyopaque, ) u64 {
        return qtc.QOpenGLTimerQuery_WaitForResult(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimerQuery_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimerquery.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimerQuery_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimerquery.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLTimerQuery_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qopengltimemonitor.html
pub const qopengltimemonitor = struct {

    /// New constructs a new QOpenGLTimeMonitor object.
    pub fn New() QtC.QOpenGLTimeMonitor {
        return qtc.QOpenGLTimeMonitor_new();
    }


    /// New2 constructs a new QOpenGLTimeMonitor object.
    pub fn New2(parent: ?*anyopaque) QtC.QOpenGLTimeMonitor {
        return qtc.QOpenGLTimeMonitor_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimeMonitor_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimemonitor.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetSampleCount(self: ?*anyopaque, sampleCount: i32) void {
        qtc.QOpenGLTimeMonitor_SetSampleCount(@ptrCast(self), sampleCount);
    }

    pub fn SampleCount(self: ?*anyopaque, ) i32 {
        return qtc.QOpenGLTimeMonitor_SampleCount(@ptrCast(self));
    }

    pub fn Create(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimeMonitor_Create(@ptrCast(self));
    }

    pub fn Destroy(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimeMonitor_Destroy(@ptrCast(self));
    }

    pub fn IsCreated(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimeMonitor_IsCreated(@ptrCast(self));
    }

    pub fn ObjectIds(self: ?*anyopaque, ) []const u8 {
        return qtc.QOpenGLTimeMonitor_ObjectIds(@ptrCast(self));
    }

    pub fn RecordSample(self: ?*anyopaque, ) i32 {
        return qtc.QOpenGLTimeMonitor_RecordSample(@ptrCast(self));
    }

    pub fn IsResultAvailable(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLTimeMonitor_IsResultAvailable(@ptrCast(self));
    }

    pub fn WaitForSamples(self: ?*anyopaque, ) []const u8 {
        return qtc.QOpenGLTimeMonitor_WaitForSamples(@ptrCast(self));
    }

    pub fn WaitForIntervals(self: ?*anyopaque, ) []const u8 {
        return qtc.QOpenGLTimeMonitor_WaitForIntervals(@ptrCast(self));
    }

    pub fn Reset(self: ?*anyopaque, ) void {
        qtc.QOpenGLTimeMonitor_Reset(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimeMonitor_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimemonitor.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QOpenGLTimeMonitor_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qopengltimemonitor.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLTimeMonitor_Delete(@ptrCast(self));
    }
};

File: binding/src/opengl/libqopengltimerquery.cpp
#include <QObject>
#include <QOpenGLTimeMonitor>
#include <QOpenGLTimerQuery>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qopengltimerquery.h>
#include "libqopengltimerquery.h"
#include "libqopengltimerquery.hxx"

QOpenGLTimerQuery* QOpenGLTimerQuery_new() {
	 return new QOpenGLTimerQuery();
}

QOpenGLTimerQuery* QOpenGLTimerQuery_new2(QObject* parent) {
	 return new QOpenGLTimerQuery(parent);
}

libqt_string QOpenGLTimerQuery_Tr(const char* s) {
	QString _ret = QOpenGLTimerQuery::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QOpenGLTimerQuery_Create(QOpenGLTimerQuery* self) {
	return self->create();
}

void QOpenGLTimerQuery_Destroy(QOpenGLTimerQuery* self) {
	self->destroy();
}

bool QOpenGLTimerQuery_IsCreated(const QOpenGLTimerQuery* self) {
	return self->isCreated();
}

uint32_t QOpenGLTimerQuery_ObjectId(const QOpenGLTimerQuery* self) {
	return self->objectId();
}

void QOpenGLTimerQuery_Begin(QOpenGLTimerQuery* self) {
	self->begin();
}

void QOpenGLTimerQuery_End(QOpenGLTimerQuery* self) {
	self->end();
}

uint64_t QOpenGLTimerQuery_WaitForTimestamp(const QOpenGLTimerQuery* self) {
	return self->waitForTimestamp();
}

void QOpenGLTimerQuery_RecordTimestamp(QOpenGLTimerQuery* self) {
	self->recordTimestamp();
}

bool QOpenGLTimerQuery_IsResultAvailable(const QOpenGLTimerQuery* self) {
	return self->isResultAvailable();
}

uint64_t QOpenGLTimerQuery_WaitForResult(const QOpenGLTimerQuery* self) {
	return self->waitForResult();
}

libqt_string QOpenGLTimerQuery_Tr2(const char* s, const char* c) {
	QString _ret = QOpenGLTimerQuery::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QOpenGLTimerQuery_Tr3(const char* s, const char* c, int n) {
	QString _ret = QOpenGLTimerQuery::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QOpenGLTimerQuery_Delete(QOpenGLTimerQuery* self) {
    delete self;
}

QOpenGLTimeMonitor* QOpenGLTimeMonitor_new() {
	 return new QOpenGLTimeMonitor();
}

QOpenGLTimeMonitor* QOpenGLTimeMonitor_new2(QObject* parent) {
	 return new QOpenGLTimeMonitor(parent);
}

libqt_string QOpenGLTimeMonitor_Tr(const char* s) {
	QString _ret = QOpenGLTimeMonitor::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QOpenGLTimeMonitor_SetSampleCount(QOpenGLTimeMonitor* self, int sampleCount) {
	self->setSampleCount(sampleCount);
}

int QOpenGLTimeMonitor_SampleCount(const QOpenGLTimeMonitor* self) {
	return self->sampleCount();
}

bool QOpenGLTimeMonitor_Create(QOpenGLTimeMonitor* self) {
	return self->create();
}

void QOpenGLTimeMonitor_Destroy(QOpenGLTimeMonitor* self) {
	self->destroy();
}

bool QOpenGLTimeMonitor_IsCreated(const QOpenGLTimeMonitor* self) {
	return self->isCreated();
}

libqt_list QOpenGLTimeMonitor_ObjectIds(const QOpenGLTimeMonitor* self) {
	return self->objectIds();
}

int QOpenGLTimeMonitor_RecordSample(QOpenGLTimeMonitor* self) {
	return self->recordSample();
}

bool QOpenGLTimeMonitor_IsResultAvailable(const QOpenGLTimeMonitor* self) {
	return self->isResultAvailable();
}

libqt_list QOpenGLTimeMonitor_WaitForSamples(const QOpenGLTimeMonitor* self) {
	return self->waitForSamples();
}

libqt_list QOpenGLTimeMonitor_WaitForIntervals(const QOpenGLTimeMonitor* self) {
	return self->waitForIntervals();
}

void QOpenGLTimeMonitor_Reset(QOpenGLTimeMonitor* self) {
	self->reset();
}

libqt_string QOpenGLTimeMonitor_Tr2(const char* s, const char* c) {
	QString _ret = QOpenGLTimeMonitor::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QOpenGLTimeMonitor_Tr3(const char* s, const char* c, int n) {
	QString _ret = QOpenGLTimeMonitor::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QOpenGLTimeMonitor_Delete(QOpenGLTimeMonitor* self) {
    delete self;
}

